src/ray/gcs/gcs_server/gcs_health_check_manager.h

```diff
@@ -0,0 +1,164 @@
+// Copyright 2022 The Ray Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <grpcpp/grpcpp.h>
+
+#include "absl/container/flat_hash_map.h"
+#include "ray/common/asio/instrumented_io_context.h"
+#include "ray/common/id.h"
+#include "ray/common/ray_config.h"
+#include "src/proto/grpc/health/v1/health.grpc.pb.h"
+
+class GcsHealthCheckManagerTest;
+
+namespace ray {
+namespace gcs {
+
+/// GcsHealthCheckManager is used to track the healthiness of the nodes in the ray
+/// cluster. The health check is done in pull based way, which means this module will send
+/// health check to the raylets to see whether the raylet is healthy or not. If the raylet
+/// is not healthy for certain times, the module will think the raylet dead.
+/// When the node is dead a callback passed in the constructor will be called and this
+/// node will be removed from GcsHealthCheckManager. The node can be added into this class
+/// later. Although the same node id is not supposed to be reused in ray cluster, this is
+/// not enforced in this class. The implementation of this class try its bes not to couple
+/// itself with GCS so we can move it to ray/common/health_check_manager.h in the future.
```

Comment:
Are we going to use it for checking health between RPCs?

Reply:
Btw, I feel like we don't really need to mention this part (`The implementation of this class try its bes not to couple...`), but it's up to you..

Reply:
I'm ok with removing the comment if you insist. Btw, I plan to move it to ray/common later.